### ISSUE TITLE
Loading a PNG bigger than 4096 causes image corruption

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#11504]
Godot 2.1.3, Godot 2.1.4
Windows 10 64 bits

Loading an image with one dimension larger than 4096 causes corruption of the image.

As soon as the size is 4097 or higher, corruption appears:
![image](https://user-images.githubusercontent.com/1311555/30769093-631526ac-a012-11e7-908d-947627bfde84.png)

**Comment by eon-s** [comment]
Isn't 4096 the limit of GLES2?  (or most GLES2 supported hardware)

**Comment by Calinou** [comment]
> Isn't 4096 the limit of GLES2? (or most GLES2 supported hardware)

Modern desktop hardware supports 8192² textures (e.g. GeForce 8 series and later) or 16384² textures (e.g. GeForce 400 series and later). This probably should apply to Godot, since it uses OpenGL on desktop platforms (since OpenGL ES is not available there; GLEW wraps OpenGL ES to OpenGL for use on desktop platforms).

**Comment by eon-s** [comment]
Yes, I meant the limits on mobile devices, which seem to be the limits of 2.x
